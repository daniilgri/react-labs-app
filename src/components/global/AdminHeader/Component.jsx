import React, { useState, useEffect, useRef } from "react";
import { faBars } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import {
  Wrapper,
  BackLink,
  MenuItem,
  Menu,
  StyledNavLink,
  MenuToggle,
} from "./styles";

const Component = () => {
  const [activeMenu, setActiveMenu] = useState(false);
  const [activeProfileSelect, setActiveProfileSelect] = useState(false);

  const profileSelectRef = useRef(null);

  const handleMenuToggleOnClick = (event) => {
    setActiveMenu((prevState) => !prevState);
  };

  const handleClickOutside = (event) => {
    if (
      activeProfileSelect &&
      profileSelectRef.current &&
      !profileSelectRef.current.contains(event.target)
    ) {
      setActiveProfileSelect(false);
    }
  };

  useEffect(() => {
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  });

  return (
    <Wrapper role="custom-dropdown">
      <MenuToggle onClick={handleMenuToggleOnClick}>
        <FontAwesomeIcon icon={faBars} />
      </MenuToggle>
      <BackLink to="/profile">Go to profile</BackLink>
      <Menu active={activeMenu}>
        <MenuItem>
          <StyledNavLink to="/admin/users">Users</StyledNavLink>
        </MenuItem>
        <MenuItem>
          <StyledNavLink exact to="/admin/films">
            Films
          </StyledNavLink>
        </MenuItem>
        <MenuItem>
          <StyledNavLink to="/admin/films/new">New film</StyledNavLink>
        </MenuItem>
      </Menu>
    </Wrapper>
  );
};

export default Component;
