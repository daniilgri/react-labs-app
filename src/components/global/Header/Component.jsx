import React, { useState, useEffect, useRef } from "react";
import PropTypes from "prop-types";
import { faBars, faUserCircle } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import {
  Wrapper,
  LogoWrapper,
  MenuItem,
  Menu,
  StyledNavLink,
  MenuToggle,
  Select,
  SelectIcon,
  OptionsList,
  Option,
  StyledLink,
  OptionLink,
} from "./styles";

const Component = ({ user, loading, error, signOutRequested }) => {
  const [activeMenu, setActiveMenu] = useState(false);
  const [activeProfileSelect, setActiveProfileSelect] = useState(false);

  const profileSelectRef = useRef(null);

  const handleMenuToggleOnClick = (event) => {
    setActiveMenu((prevState) => !prevState);
  };

  const handleProfileSelectOnClick = (event) => {
    setActiveProfileSelect((prevState) => !prevState);
  };

  const handleClickOutside = (event) => {
    if (
      activeProfileSelect &&
      profileSelectRef.current &&
      !profileSelectRef.current.contains(event.target)
    ) {
      setActiveProfileSelect(false);
    }
  };

  const handleDefaultCloseOptionOnClick = (event) => {
    setActiveProfileSelect(false);
  };

  const handleSignOutOptionOnClick = (event) => {
    event.preventDefault();
    signOutRequested();
    setActiveProfileSelect(false);
  };

  useEffect(() => {
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  });

  return (
    <Wrapper role="custom-dropdown">
      <MenuToggle onClick={handleMenuToggleOnClick}>
        <FontAwesomeIcon icon={faBars} />
      </MenuToggle>
      <LogoWrapper>Logo</LogoWrapper>
      <Menu active={activeMenu}>
        <MenuItem>
          <StyledNavLink to="/">All films</StyledNavLink>
        </MenuItem>
      </Menu>
      {!loading && (
        <Select active={activeMenu} ref={profileSelectRef}>
          <SelectIcon onClick={handleProfileSelectOnClick}>
            <FontAwesomeIcon icon={faUserCircle} />
          </SelectIcon>
          <OptionsList active={activeProfileSelect}>
            {user ? (
              <React.Fragment>
                <OptionLink
                  to="/profile"
                  onClick={handleDefaultCloseOptionOnClick}
                >
                  Profile
                </OptionLink>
                <OptionLink
                  to="/profile/orders"
                  onClick={handleDefaultCloseOptionOnClick}
                >
                  Orders
                </OptionLink>
                <Option onClick={handleSignOutOptionOnClick}>Sign out</Option>
              </React.Fragment>
            ) : (
              <React.Fragment>
                <OptionLink
                  to="/login"
                  onClick={handleDefaultCloseOptionOnClick}
                >
                  Login
                </OptionLink>
                <OptionLink
                  to="/signup"
                  onClick={handleDefaultCloseOptionOnClick}
                >
                  Sign up
                </OptionLink>
              </React.Fragment>
            )}
          </OptionsList>
        </Select>
      )}
    </Wrapper>
  );
};

Component.defaultProps = {
  user: null,
  loading: false,
  error: "",
  signOutRequested: () => {},
};

Component.propTypes = {
  user: PropTypes.object,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string,
  signOutRequested: PropTypes.func.isRequired,
};

export default Component;
