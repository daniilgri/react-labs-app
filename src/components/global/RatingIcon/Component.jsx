import React, { useMemo } from "react";
import PropTypes from "prop-types";
import StarIcon from "../StarIcon";
import { Wrapper } from "./styles";

const Component = ({ index, rating, hoverRating, onMouseEnter, onMouseLeave, onSaveRating }) => {
  const fill = useMemo(() => {
    if (hoverRating >= index) {
      return "yellow";
    }
    if (!hoverRating && rating >= index) {
      return "yellow";
    }
    return "none";
  }, [rating, hoverRating, index]);

  const handleMouseEnter = () => {
    onMouseEnter(index);
  };

  const handleMouseLeave = () => {
    onMouseLeave();
  };

  const handleIconOnClick = () => {
    onSaveRating(index);
  };

  return (
    <Wrapper
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      onClick={handleIconOnClick}
    >
      <StarIcon fill={fill} />
    </Wrapper>
  );
};

Component.propTypes = {
  index: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
  hoverRating: PropTypes.number.isRequired,
  onMouseEnter: PropTypes.func.isRequired,
  onMouseLeave: PropTypes.func.isRequired,
  onSaveRating: PropTypes.func.isRequired,
};

export default Component;
