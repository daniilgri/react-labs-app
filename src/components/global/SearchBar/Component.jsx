import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";
import { useFormik } from "formik";

import { Wrapper, InputWrapper, Input, IconWrapper } from "./styles";

const Component = ({ onSet, value }) => {
  const formik = useFormik({
    initialValues: {
      query: value,
    },
    onSubmit: values => {
      onSet(values);
    },
  });

  return (
    <Wrapper onSubmit={formik.handleSubmit}>
      <InputWrapper>
        <Input
          placeholder="Search..."
          name="query"
          id="filmsBoardSearchBar"
          onChange={formik.handleChange}
          value={formik.values.query}
        />
        <IconWrapper type="submit">
          <FontAwesomeIcon icon={faSearch} />
        </IconWrapper>
      </InputWrapper>
    </Wrapper>
  );
};

Component.defaultProps = {
  value: "",
};

Component.propTypes = {
  value: PropTypes.string,
  onSet: PropTypes.func.isRequired,
};

export default Component;
