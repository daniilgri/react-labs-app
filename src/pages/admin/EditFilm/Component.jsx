import React from "react";
import { useParams } from "react-router-dom";

import AdminHeader from "../../../components/global/AdminHeader";
import Form from "../../../components/related/admin/EditFilm/Form";
import { Wrapper } from "./styles";

const Component = () => {
  const { id } = useParams();

  return (
    <>
      <AdminHeader />
      <Wrapper>
        <Form filmId={id} />
      </Wrapper>
    </>
  );
};

export default Component;
