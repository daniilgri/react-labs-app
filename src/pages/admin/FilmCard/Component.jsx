import React from "react";
import { useParams } from "react-router-dom";

import AdminHeader from "../../../components/global/AdminHeader";
import Card from "../../../components/related/admin/FilmCard/Card";
import Subscribers from "../../../components/related/admin/FilmCard/Subscribers";
import { Wrapper } from "./styles";

const Component = () => {
  const { id } = useParams();

  return (
    <>
      <AdminHeader />
      <Wrapper>
        <Card filmId={id} />
        <Subscribers filmId={id} />
      </Wrapper>
    </>
  );
};

export default Component;
