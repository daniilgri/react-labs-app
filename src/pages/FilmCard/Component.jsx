import React from "react";
import { useParams } from "react-router-dom";

import Header from "../../components/global/Header";
import Card from "../../components/related/FilmCard/Card";
import { Wrapper } from "./styles";

const Component = () => {
  const { id } = useParams();

  return (
    <React.Fragment>
      <Header />
      <Wrapper>
        <Card filmId={id} />
      </Wrapper>
    </React.Fragment>
  );
};

export default Component;
