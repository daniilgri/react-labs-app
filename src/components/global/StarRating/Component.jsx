import React, { useState } from "react";
import PropTypes from "prop-types";
import { v4 as uuidv4 } from "uuid";

import RatingIcon from "../RatingIcon";
import { Wrapper } from "./styles";

const Component = ({ count, rate, onRateSet }) => {
  const [hoverRating, setHoverRating] = useState(0);

  const onMouseEnter = index => {
    setHoverRating(index);
  };

  const onMouseLeave = () => {
    setHoverRating(0);
  };

  const onSaveRating = index => {
    onRateSet(index);
  };

  return (
    <Wrapper>
      {Array.from(Array(count).keys()).map(index => {
        return (
          <RatingIcon
            key={uuidv4()}
            index={index + 1}
            rating={rate}
            hoverRating={hoverRating}
            onMouseEnter={onMouseEnter}
            onMouseLeave={onMouseLeave}
            onSaveRating={onSaveRating}
          />
        );
      })}
    </Wrapper>
  );
};

Component.propTypes = {
  count: PropTypes.number.isRequired,
  rate: PropTypes.number.isRequired,
  onRateSet: PropTypes.func.isRequired,
};

export default Component;
