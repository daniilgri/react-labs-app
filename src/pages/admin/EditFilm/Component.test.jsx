import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

import Component from "./Component";

const mockStore = configureStore([]);

describe("EditFilm component", () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      editFilm: {
        loading: false,
        error: "",
        film: {
          title: "",
          description: "",
          tags: [],
          image: "",
          rates: [],
          screeningDates: [],
          ticketPrice: "",
        },
      },
      addScreeningDateModal: {
        open: false,
      },
    });
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Component />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
