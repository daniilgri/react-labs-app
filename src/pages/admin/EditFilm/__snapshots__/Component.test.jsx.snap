// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditFilm component renders correctly 1`] = `
Array [
  <nav
    className="sc-bdfBwQ grEbaU"
  >
    <div
      className="sc-jSgupP NXinf"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-w-14 "
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <a
      className="sc-gsTCUz fKNCie"
      href="/profile"
      onClick={[Function]}
    >
      Go to profile
    </a>
    <ul
      className="sc-dlfnbm uDvuY"
    >
      <li
        className="sc-hKgILt kYjJwg"
      >
        <a
          aria-current={null}
          className="sc-eCssSg iyHtsY"
          href="/admin/users"
          onClick={[Function]}
        >
          Users
        </a>
      </li>
      <li
        className="sc-hKgILt kYjJwg"
      >
        <a
          aria-current={null}
          className="sc-eCssSg iyHtsY"
          href="/admin/films"
          onClick={[Function]}
        >
          Films
        </a>
      </li>
      <li
        className="sc-hKgILt kYjJwg"
      >
        <a
          aria-current={null}
          className="sc-eCssSg iyHtsY"
          href="/admin/films/new"
          onClick={[Function]}
        >
          New film
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="sc-aemoO hUrqFk"
  >
    <form
      className="sc-clsHhM gcwzZZ"
      onKeyDown={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="sc-fbkhIv bNwILm"
      >
        <h2
          className="sc-GqfZa danEqG"
        >
          Edit film - 
          
        </h2>
         
        <a
          className="sc-fWSCIC gQRrhx"
          href="/admin/film/undefined"
          onClick={[Function]}
        >
          Go to film card
        </a>
      </div>
      <div
        className="sc-dwfUOf fUAEWa"
      >
        <div
          className="sc-crrsfI cJqrZf"
        >
          <div
            className="sc-dQppl hnKiiC"
          >
            <h2
              className="sc-bqyKva hDaAzP"
            >
              Add description info
            </h2>
          </div>
          <div
            className="sc-kstrdz rKYPv"
          >
            <div
              className="sc-hBEYos dIIpzz"
            >
              <label
                className="sc-fodVxV japWFt"
                htmlFor="title"
              >
                Title
              </label>
              <input
                className="sc-fFubgz gdURtZ"
                id="title"
                name="title"
                onChange={[Function]}
                placeholder="title"
                type="text"
                value=""
              />
            </div>
            <div
              className="sc-hBEYos dIIpzz"
            >
              <label
                className="sc-fodVxV japWFt"
                htmlFor="description"
              >
                Description
              </label>
              <textarea
                className="sc-bkzZxe cqpWQA"
                id="description"
                name="description"
                onChange={[Function]}
                placeholder="..."
                type="text"
                value=""
              />
            </div>
            <div
              className="sc-dIUggk izSkSw"
            >
              <div
                className="sc-iBPRYJ jHiIsc"
              >
                <ul
                  className="sc-fubCfw pTHLR"
                />
                <input
                  className="sc-kEjbxe gmminW"
                  id="tagInput"
                  name="tagInput"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Type tag"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-hHftDr kRIWcI"
        >
          <div
            className="sc-dmlrTW jkOiiX"
          >
            <h2
              className="sc-kfzAmx efuZiy"
            >
              Add media info
            </h2>
          </div>
          <div
            className="sc-fKFyDc kVEVXy"
          >
            <img
              alt="preview"
              className="sc-iJuUWI lcrZAx"
              src=""
            />
            <div
              className="sc-bBXqnf haPuhk"
            >
              <label
                className="sc-iwyYcG gvGGaq"
                htmlFor="imageAsFile"
              >
                Image
              </label>
              <input
                className="sc-cxFLnm frsgui"
                id="imageAsFile"
                name="imageAsFile"
                onChange={[Function]}
                type="file"
              />
            </div>
          </div>
        </div>
        <div
          className="sc-jNMdTA bVYHRQ"
        >
          <div
            className="sc-dOSReg chhgLj"
          >
            <h2
              className="sc-bBrOnJ hqmTqI"
            >
              Formal
            </h2>
          </div>
          <div
            className="sc-cOajty fxnhiZ"
          >
            <div
              className="sc-AzgDb fGbrWD"
            >
              <label
                className="sc-khAkjo dvAqag"
                htmlFor="screeningDates"
              >
                Screening dates:
              </label>
              <div
                className="sc-gTgzIj gSQLgx"
              >
                <button
                  className="sc-laRPJI fKFCwY"
                  onClick={[Function]}
                  type="button"
                >
                  Add
                </button>
              </div>
            </div>
            <div
              className="sc-AzgDb fGbrWD"
            >
              <label
                className="sc-khAkjo dvAqag"
                htmlFor="ticketPrice"
              >
                Ticket price
              </label>
              <input
                className="sc-hTZhsR kiVfIc"
                id="ticketPrice"
                name="ticketPrice"
                onChange={[Function]}
                placeholder="price"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div />
      </div>
      <button
        className="sc-TmcTc iaQIkC"
        type="submit"
      >
        Publish
      </button>
    </form>
  </div>,
]
`;
